source/common/quic/envoy_quic_dispatcher.cc

```diff
@@ -126,6 +126,23 @@ std::unique_ptr<quic::QuicSession> EnvoyQuicDispatcher::CreateQuicSession(
   return quic_session;
 }
 
+bool EnvoyQuicDispatcher::OnFailedToDispatchPacket(
+    const quic::ReceivedPacketInfo& received_packet_info) {
+  if (hot_restart_forward_packet_ == nullptr) {
+    // If we're not hot restarting, do the default behavior.
+    return quic::QuicDispatcher::OnFailedToDispatchPacket(received_packet_info);
+  }
+  // We are hot restarting, forward the packet to the new instance.
+  Network::UdpRecvData packet;
+  packet.addresses_.local_ = quicAddressToEnvoyAddressInstance(received_packet_info.self_address);
+  packet.addresses_.peer_ = quicAddressToEnvoyAddressInstance(received_packet_info.peer_address);
+  packet.buffer_ = std::make_unique<Buffer::OwnedImpl>(received_packet_info.packet.AsStringPiece());
+  packet.receive_time_ = MonotonicTime(
+      std::chrono::microseconds{received_packet_info.packet.receipt_time().ToDebuggingValue()});
```

Comment:
ToDebuggingValue() is not for production. Can you revert engineer how Envoy convert these structs during packet reading to calculate the receipt time if you choose not to avoid the conversion back and forth?